core/string/scan_spec.rb

```diff
@@ -84,6 +84,11 @@
     a = "hello".taint.scan(/./)
     a.each { |m| m.tainted?.should be_true }
   end
+
+  # jruby/jruby#5513
+  it "does not raise any errors when passed a multi-byte string" do
+    "あああaaaあああ".scan("あああ").should == ["あああ", "あああ"]
```

Comment:
It would be interesting to have different characters and not 3 times the same, as well as testing a bit more edge cases (e.g., a partial match in the middle of ASCII characters).
This is a good start though.